src/main/java/racingcar/domain/Cars.java

```diff
@@ -0,0 +1,79 @@
+package racingcar.domain;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import racingcar.domain.strategy.MoveStrategy;
+
+public class Cars {
+
+    private final List<Car> cars;
+
+    public Cars(List<Car> cars) {
+        verifyMinSize(cars);
+        verifyDuplicate(cars);
+        this.cars = new ArrayList<>(cars);
+    }
+
+    public static Cars of(List<String> nameList) {
+        List<Car> cars = new ArrayList<>();
+        for (String name : nameList) {
+            cars.add(new RacingCar(name));
+        }
+        return new Cars(cars);
+    }
+
+    private void verifyMinSize(List<Car> cars) {
+        if (cars.isEmpty()) {
+            throw new IllegalStateException();
+        }
+    }
+
+    private void verifyDuplicate(List<Car> cars) {
+        Set<String> carNames = cars.stream()
+            .map(Car::getName)
+            .collect(Collectors.toSet());
+
+        if(carNames.size() != cars.size()){
```

Comment:
코드 자동 정렬은 습관적으로 진행해주세요! 🙏 
```suggestion
        if (carNames.size() != cars.size()) {
```